openqaoa/optimizers/result.py

```diff
@@ -143,3 +143,53 @@ def plot_cost(self, figsize=(10, 8), label='Cost', linestyle='--', color='b', ax
         ax.set_title('Cost history')
 
         return
+
+    def plot_probabilities(self, figsize = (10,8),label='Probability distribution',color='tab:blue', ax=None):
+
+        """
+        Helper function to plot the probabilities corresponding to each basis states (with prob != 0) obtained from the optimized result
+
+        Parameters
+        ----------
+        figsize: `tuple`
+            The size of the figure to be plotted. Defaults to (10,8).
+        label: `str`
+            The label of the cost line, defaults to 'Probability distribution'.
+        color: `str`
+            The color of the line. Defaults to 'tab:blue'.
+        ax: 'matplotlib.axes._subplots.AxesSubplot'
+            Axis on which to plot the graph. Deafults to None
+        """
+
+        outcome = self.optimized['optimized measurement outcomes']
+
+        # converting to counts dictionary if outcome is statevector
+        if type(outcome) == type(np.array([])):
+            outcome = self.get_counts(outcome)
+            # setting norm to 1 since it might differ slightly for statevectors due to numerical preicision
+            norm = np.float64(1)
+        else: 
+            # needed to be able to divide the tuple by 'norm'
+            norm = np.float64(sum(outcome.values())) 
+
+        # sorting dictionary by increasing binary value (i.e. keys) and unzipping the states and counts
+        outcome_list = sorted(outcome.items())
+        states, counts = zip(*outcome_list)
+
+        # normalizing to obtain probabilities
+        probs = counts/norm
+
+        # formatting labels
+        labels = [r'$\left|{}\right>$'.format(state) for state in states]
+
+        if ax is None:
+            fig, ax = plt.subplots(figsize=figsize)
+
+        ax.bar(labels,probs, color=color)
+        ax.set_xlabel('Eigen-State')
+        ax.set_ylabel('Probability')
+        ax.set_title(label)
+        ax.tick_params(axis='x', labelrotation = 70)
+        ax.grid(True, axis='y', linestyle='--')
```

Comment:
Hmm. When I run this method in a jupyter notebook, the plot doesnt show until the user does `matplotlib.pyplot.show()` . Is this the intended behaviour?

Reply:
it works well for me, could you tell me more about the instance you tried it on?

<img width="1440" alt="Screenshot 2022-09-27 at 23 02 57" src="https://user-images.githubusercontent.com/57567043/192691747-04ec7b51-1506-4c8c-9330-d0250b9b7ccc.png">
![Screenshot 2022-09-27 at 23 02 57 2](https://user-images.githubusercontent.com/57567043/192691760-1d02f2d5-5902-4b1b-b732-3a69292ef67c.png)

Reply:
I have done the changes, let me know what you think :)

Reply:
Can we check your two versions of mathplotlib?

Reply:
I'm using '3.5.3'

Reply:
I'm using `3.4.3`

But I think i figured it out. It seems that I wasn't running `%matplotlib inline` in my notebook.
If that magic method isnt used, the user has to specifically request `pyplot` to show the plot with `matplotlib.pyplot.show()`.

Reply:
perhaps you won't need if you include `plt.show()` in the body of the function, no?

Reply:
Yes. As long as `plt.show()` is ran within the function, we wont need to have the user to run it by himself.

Reply:
I see. I will add it then! Should I also add it to the `plot_cost` method?

Reply:
Looks like it might just be some versioning difference where this is an issue. We've decided to just integrate this version of the PR without the implementation of `plt.show()` :)